segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -127,35 +131,44 @@
 
     @Override
     public void close() {
-        val w = this.writer.get();
-        if (w != null) {
-            w.close();
-            this.writer.set(null);
+        writer.cancel(true);
+
+        if (Futures.isSuccessful(writer)) {
+            val w = this.writer.join();
+            if (w != null) {
+                w.close();
+            }
         }
 
-        this.clientFactory.close();
+        if (clientFactory != null) {
+            this.clientFactory.close();
+        }
         this.cacheCleanup.cancel(true);
     }
-
+    
     private void bootstrapRequestWriters(EventStreamClientFactory clientFactory, ScheduledExecutorService executor) {
-        // Starting with initial delay, in case request stream has not been created, to give it time to start
-        // However, we have this wrapped in consumeFailure which means the creation of writer will be retried.
-        // We are introducing a delay to avoid exceptions in the log in case creation of writer is attempted before
-        // creation of requeststream.
-        executor.schedule(
-                () -> Retry.indefinitelyWithExpBackoff(100, 10, 10000, this::handleBootstrapException)
-                        .runInExecutor(() -> bootstrapOnce(clientFactory), executor),
-                10, TimeUnit.SECONDS);
+        AtomicReference<EventStreamWriter<AutoScaleEvent>> w = new AtomicReference<>();
+
+        Futures.completeAfter(() -> Retry.indefinitelyWithExpBackoff(100, 10, 10000, this::handleBootstrapException)
+                                         .runInExecutor(() -> bootstrapOnce(clientFactory, w), 
+                                                 executor).thenApply(v -> w.get()), writer);
     }
 
     private void handleBootstrapException(Throwable e) {
         log.warn("Unable to create writer for requeststream: {}.", LoggerHelpers.exceptionSummary(log, e));
```

Comment:
If `startInitWriter` is `false` then we would be logging this line repeatedly. 
This would clutter the logs with WARN messages. (including standalone) Can we avoid it?